bin/oneoff/backfill_data/backfill_script_ids_in_teacher_feedbacks.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+# Backfill existing TeacherFeedbacks to set script_id based on script_level_id.
+
+require 'optparse'
+
+$options = {actually_update: false}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-u', '--actually-update', 'Actually perform the update.') do
+    $options[:actually_update] = true
+  end
+  opts.on('-h', '--help', 'Add -u to perform the update.') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Called with options: #{$options}"
+
+require_relative '../../../dashboard/config/environment'
+
+$stdout.sync = true
+
+def feedbacks_without_script_id
+  TeacherFeedback.where(script_id: nil)
+end
+
+def puts_count
+  puts "There are #{feedbacks_without_script_id.count} feedbacks without a script_id"
+end
+
+def update_script_ids
+  puts "starting backfilling script_ids"
+  feedbacks_without_script_id.find_in_batches(batch_size: 1000) do |batch|
+    print '.'
+    ActiveRecord::Base.transaction do
```

Comment:
Do we need each batch of 1000 `feedbacks_without_script_id` to succeed or fail as a unit? If not, I suggest starting the transaction within the inner loop, and have a single transaction for each teacher_feedback.

Reply:
> Do we need each batch of 1000 feedbacks_without_script_id to succeed or fail as a unit?

I would say "yes". this rollback should only happen if we did not specify `actually_update`, in which case we want all DB updates to be rolled back. if any other kind of error occurs, we report it and continue.